cmd/kubeadm/app/cmd/init.go

```diff
@@ -309,6 +309,23 @@ func (i *Init) Run(out io.Writer) error {
 
 	adminKubeConfigPath := filepath.Join(kubeConfigDir, kubeadmconstants.AdminKubeConfigFileName)
 
+	if !i.dryRun {
+		// read the kubeadm log file and write a new one if it doesn't exist
+		data, err := kubeadmutil.ReadKubeadmLogFile()
+		if err != nil {
+			if err := kubeadmutil.WriteKubeadmLogFile("kubeadm init"); err != nil {
+				glog.Fatalf("[init] cannot write kubeadm log file: %v", err)
```

Comment:
Please, replace all occurrences of `glog.Fatal` with `glog.Error`, also return the error to the caller function.

`glog.Fatal` invokes `os.Exit` and does not allow for any kind of cleanups and error handling.